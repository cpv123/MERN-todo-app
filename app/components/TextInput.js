import React, { Component } from 'react';
import TextField from 'material-ui/TextField';

export default class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            input: ""
        }
    };

    onChange = event => {
        console.log(this.state.text);
        const { value: text } = e.target;
        this.setState({
            text,
        });
    };

    onKeyDown = e => {
        const { value } = e.target;
        if (e.which === 13) {
            //this.props.onSubmit(value);
            console.log('Enter')
            this.setState({
                text: "",
            });
        }
    };

    render() {
        return (
            <TextField
                floatingLabelText="What do you need to do?"
                value={this.state.input}
                onChange={this.onChange}
                onKeyDown={this.onKeyDown}
            />
        );
    } 
}