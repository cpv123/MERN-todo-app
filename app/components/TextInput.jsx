import React from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';

const TextInput = ({ value, onChange, label }) => (
  <TextField
    floatingLabelText={label}
    value={value}
    onChange={onChange}
    style={{ fontSize: 20, marginBottom: 10 }}
  />
);

TextInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
};

export default TextInput;
