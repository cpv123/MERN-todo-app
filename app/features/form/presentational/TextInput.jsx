import React from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';

const TextInput = ({ value, onChange }) => (
  <TextField
    floatingLabelText="What do you need to do?"
    value={value}
    onChange={onChange}
    style={{ fontSize: 20, marginBottom: 10 }}
  />
);

TextInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;
