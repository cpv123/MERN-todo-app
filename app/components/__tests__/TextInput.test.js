/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import TextInput from '../TextInput.jsx';

describe('the TextInput component', () => {
  it('renders correctly', () => {
    const mockProps = { value: "Test input" };
    const wrapper = shallow(<TextInput {...mockProps}/>);
    expect(wrapper).toMatchSnapshot();
  });
});