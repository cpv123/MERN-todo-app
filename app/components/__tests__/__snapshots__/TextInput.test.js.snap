// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TextInput component renders correctly 1`] = `
<TextField
  disabled={false}
  floatingLabelFixed={false}
  fullWidth={false}
  multiLine={false}
  rows={1}
  style={
    Object {
      "fontSize": 20,
      "marginBottom": 10,
    }
  }
  type="text"
  underlineShow={true}
  value="Test input"
/>
`;
